serde-api/src/main/java/io/micronaut/serde/config/SerdeConfiguration.java

```diff
@@ -59,6 +59,17 @@ public interface SerdeConfiguration {
     @Bindable(defaultValue = "SECONDS")
     NumericTimeUnit getNumericTimeUnit();
 
+    /**
+     * Control whether to use legacy behavior for writing byte arrays. When set to {@code true} (the
+     * default in serde 2.x), byte arrays will always be written as arrays of numbers. When set to
+     * {@code false}, the encoding may be format-specific instead, and will be a base64 string for
+     * JSON.
+     *
+     * @return Whether to use legacy byte array writing behavior
+     */
+    @Bindable(defaultValue = "true")
+    boolean isWriteLegacyByteArrays();
```

Comment:
Rename to `isWriteByteArraysAsBinary` and flip logic. Makes more logical sense from a configuration perspective and from the API perspective with the encodeBinary/decodeBinary methods.

Reply:
that's also a bit confusing though, because for json it doesnt write the byte array as binary, it writes it as base64. the encode/decode method is called that because the data is binary, even though the serialized form isnt necessarily

Reply:
yeah just not keen on this name and including `legacy` in configuration values. It should describe what it does.

Reply:
how about isWriteBinaryAsArray